components/ProfilePage/NotificationSettingsModal.tsx

```diff
@@ -1,25 +1,60 @@
-import { useState } from "react"
+import { Dispatch, SetStateAction } from "react"
 import type { ModalProps } from "react-bootstrap"
 import Dropdown from "react-bootstrap/Dropdown"
-import { Button, Col, Image, Modal, Stack } from "../bootstrap"
+import { useForm } from "react-hook-form"
+import { Button, Col, Form, Image, Modal, Stack } from "../bootstrap"
+import { ProfileHook } from "../db"
 import styles from "./NotificationSettingsModal.module.css"
 
+type UpdateProfileData = {
+  private: string
+  notification: string
+  notificationActive: string
+}
+
+type Props = Pick<ModalProps, "show" | "onHide"> & {
+  actions: ProfileHook
+  onSettingsModalClose: () => void
+  notifications: string
+  setNotifications: Dispatch<SetStateAction<"Daily" | "Weekly" | "Monthly">>
+  notificationsEnabled: string
+  setNotificationsEnabled: Dispatch<SetStateAction<"On" | "">>
+  profileSettings: string
```

Comment:
Can we name this `isProfilePublic` and make it a boolean?

Reply:
definitely, done